mm2src/mm2_main/tests/mm2_tests/tron_tests.rs

```diff
@@ -0,0 +1,512 @@
+//! TRON activation integration tests
+//!
+//! Run with: cargo test --test mm2_tests_main --features tron-network-tests tron_
+
+use coins::eth::tron::TronAddress;
+use common::block_on;
+use mm2_test_helpers::for_tests::{
+    account_balance, enable_trx, get_new_address, get_passphrase, task_enable_trx, task_enable_trx_result, trx_conf,
+    MarketMakerIt, Mm2TestConf, Mm2TestConfForSwap, TRON_NILE_NODES,
+};
+use mm2_test_helpers::structs::{
+    Bip44Chain, EnableCoinBalanceMap, EthWithTokensActivationResult, HDAccountAddressId, TaskEnableError,
+};
+
+/// Test mnemonic for used-but-zero-balance scenario.
+/// Index 0: TSqB9tqfaQ1DYSdMCbVSLPzQsaNVjeu9hq (funded ~1777.8 TRX)
+/// Index 2: TPoJwueR4xfZCXuQTYqem4edQgoM3uV78n (0 balance but has tx history)
+const TRON_USED_ZERO_BALANCE_PASSPHRASE: &str =
+    "top wonder island doctor gesture velvet local media begin impose soccer radar";
+
+/// Test TRX activation works via enable_eth_with_tokens (immediate mode).
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn test_trx_activation_immediate() {
+    let passphrase = get_passphrase(&".env.client", "BOB_PASSPHRASE").unwrap();
+    let coins = serde_json::json!([trx_conf()]);
+    let conf = Mm2TestConf::seednode(&passphrase, &coins);
```

Comment:
nit: would be nice if we abandon using these `.env.client` thing and use the hardcoded test seeds.